fabric-item-api-v1/src/testmod/java/net/fabricmc/fabric/test/item/CustomDamageTest.java

```diff
@@ -55,5 +55,11 @@ public Text getName(ItemStack stack) {
 			int v = stack.getOrCreateNbt().getInt("weird");
 			return super.getName(stack).shallowCopy().append(" (Weird Value: " + v + ")");
 		}
+
+		@Override
+		public ItemStack getRecipeRemainder(ItemStack stack) {
+			stack.setDamage(stack.getDamage() + 1);
+			return stack;
```

Comment:
Item should disappear if has the max damage.
```suggestion
			if (stack.getDamage() < stack.getMaxDamage()) {
				ItemStack moreDamaged = stack.copy();
				moreDamaged.setDamage(stack.getDamage() + 1);
				return moreDamaged;
			}

			return ItemStack.EMPTY;
```